src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -13,11 +13,113 @@ public enum DuplicatePropertyNameHandling
         Ignore = 1,
         Error = 2,
     }
+    public partial class JsonArray : System.Text.Json.JsonNode, System.Collections.Generic.ICollection<System.Text.Json.JsonNode>, System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode>, System.Collections.Generic.IList<System.Text.Json.JsonNode>, System.Collections.Generic.IReadOnlyCollection<System.Text.Json.JsonNode>, System.Collections.Generic.IReadOnlyList<System.Text.Json.JsonNode>, System.Collections.IEnumerable
+    {
+        public JsonArray() { }
+        public JsonArray(System.Collections.Generic.IEnumerable<bool> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<byte> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<decimal> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<double> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<short> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<int> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<long> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<sbyte> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<float> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<string> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<ushort> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<uint> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<ulong> values) { }
+        public int Count { get { throw null; } }
+        public bool IsReadOnly { get { throw null; } }
+        public System.Text.Json.JsonNode this[int idx] { get { throw null; } set { } }
+        public override System.Text.Json.JsonValueKind ValueKind { get { throw null; } }
+        public void Add(bool value) { }
```

Comment:
Are all these overloads necessary? I think `JsonNode` has implicit conversions from primitive types. If that is the case calling `Add(JsonNode value)` with `Add("hello world")` will do the work for you.

Reply:
Same question for ctor with IEnumerables, and Insert overloads.

Reply:
All primitive types have implicit conversions, but `JsonNode` itself does not, so it wouldn't work without overloads.

Reply:
Also, I believe the point of having all the overloads is so that you can use collection instantiation like:

```c#
var jsonObject = new JsonObject
{
  "SomeBool": true,
  "SomeString": "hello",
  "SomeNumber": 56.7,
  "SomeArray": new JsonArray
  {
    { false, "goodbye", 23 }
  }
};
```
We really only need one new statement above given that we  have all the overloads for JsonObject and JsonString.

Reply:
I think James meant that it all (including collection instantiation) should work even without overloads. But implicit conversions on JsonString, JsonNumber and JsonBoolean aren't sufficient. As I remember, at the beginning we tried the approach with implicit operators on JsonNode as well, but it seemed to require too many hacks to make it do all the magic (and it might have been too magical for users).

Reply:
What hacks are required? Implicit conversions on JToken has worked well for Newtonsoft.Json

Reply:
I found it - the hacks were needed because we had a JsonNode(Array) implicit conversion, and we needed to check if we want to create JsonObject or JsonArray from it. But in current version we anyway resigned from Array. The main downside of implicit operators would be then the magical behavior for users - they may experience conversion happening in places they wouldn't expect. And most of them would not know what is happening. Quoting @joperezr : "there are more places that expose JsonNode than more concrete types, so basically having implicit operators on JsonNode for everything might prove that we end up boxing things we don't really intend to".